src/transformStateWithClones.js

```diff
@@ -6,8 +6,66 @@
  *
  * @return {Object[]}
  */
-function transformStateWithClones(state, actions) {
-  // write code here
+
+const state = {
+  foo: 'bar',
+  name: 'Jim',
+  another: 'one',
+};
+
+function transformStateWithClones(currentState, actions) {
+  const history = [];
+  let clonedState = Object.assign({}, currentState);
+
+  actions.forEach((obj) => {
+    if (obj.type === 'clear') {
+      clonedState = {};
+      history.push({});
```

Comment:
You should not push an empty object to the history array inside the 'clear' case. Instead, push the `clonedState` after the `switch` statement at the end of each loop cycle. This will ensure that the same action is not repeated and the history array contains the correct state clones. [DONT REPEAT YOURSELF]